livedata-ktx/src/main/java/com/shopify/livedataktx/SupportMediatorLiveData.kt

```diff
@@ -28,28 +28,73 @@ import android.arch.lifecycle.LifecycleOwner
 import android.arch.lifecycle.MediatorLiveData
 import android.arch.lifecycle.Observer
 
-open class SupportMediatorLiveData<T>(internal val isSingle: Boolean = false, private val versionProvider: (() -> Int)? = null) : MediatorLiveData<T>() {
+private typealias ObserverWrappers<T> = MutableList<Pair<Observer<T>, Observer<T>>>
```

Comment:
I feel there's an opportunity to make the fix a fair bit simpler by using a Map<O,W> here?